src/components/Listings/index.js

```diff
@@ -1,47 +1,119 @@
-import React, { useState, useMemo } from "react"
+import React, { useMemo, useReducer } from "react"
 import { Link } from "gatsby"
 import { connectToSpreadsheet } from "react-google-sheet-connector"
 import classnames from "classnames"
 import { useDebounce } from "use-debounce"
 
-import { NEEDS_SHEET_NAME } from "../../utils/listingUtils"
+import {
+  NEEDS_SHEET_NAME,
+  NEED_TYPES,
+  NEEDS_SHEET_COLUMN_INDICES,
+} from "../../utils/listingUtils"
 import useTextSearch from "../../utils/useTextSearch"
 import ListingResults from "../ListingResults"
 import cs from "./styles.module.css"
 
 const FULL_TEXT_SEARCH_KEYS = ["name"]
 
+/**
+ * A reducer which parses a row from the sheet and adds the needs data to the array of all needs
+ * TODO: implement all types and parse metadata for each card
+ * @param {*} result
+ * @param {*} row
+ * @param {*} index
+ */
+function parseRow(result, row, index) {
+  const rowNeeds = []
+  if (row[NEEDS_SHEET_COLUMN_INDICES.isFinancialNeed]) {
```

Comment:
Since there's also a question on the form about "how widely" they want their info spread, maybe we should check on that here too, and not parse the row if they want their info private?

Reply:
@mewtaylor can we have a column details for that? It's easy enough to do but we need to know what the conditional logic is for when a need should be shown or hidden. This would probably be at the "row" level meaning an entire submission would either be shown or hidden.

Reply:
There is a column for this

Reply:
ok I opened #13 so we can keep track of that requirement  / work